src/TLuaInterpreter.cpp

```diff
@@ -14229,6 +14237,11 @@ void TLuaInterpreter::setMatches(lua_State* L)
             lua_pushstring(L, (*it).c_str());
             lua_settable(L, -3);
         }
+        for (auto [name, capture] : mCapturedNameGroups) {
+            lua_pushstring(L, name.toUtf8().constData());
```

Comment:
Does the PCRE need/require the names to be unique? If not there will be problems here.

Reply:
Good point but then you can just fallback on `matches[n]` which is still there

Reply:
It requires names to be unique, so result should be equal to not valid pattern, but will check thatl

Reply:
![image](https://user-images.githubusercontent.com/3740628/106811811-2c907600-666f-11eb-9ede-4cecf9d3b7af.png)

I can confirm.

Reply:
:thinking: How come the names didn't show up jn the error message? I guess that must be just the way `libpcre` does it...